common/datavalues/src/data_value.rs

```diff
@@ -376,7 +376,7 @@ impl fmt::Display for DataValue {
         match self {
             DataValue::Null => write!(f, "NULL"),
             DataValue::Boolean(v) => write!(f, "{}", v),
-            DataValue::Float64(v) => write!(f, "{}", v),
+            DataValue::Float64(v) => write!(f, "{:?}", v),
```

Comment:
Why the `{}` not work?

Reply:
`1.0` is formated to `1`. 

So `1.0 + 1` there will be duplicate column names, but their column types are different.

Reply:
Is it related to #5106 ?

Reply:
Yes, it fixed #5106 , the tests case are added.